packages/core/src/lib/session.ts

```diff
@@ -41,6 +42,30 @@ export const assignInteractionResults = async (
   ctx.body = { redirectTo, ts };
 };
 
+export const checkSessionHealth = async (
+  ctx: Context,
+  provider: Provider,
+  tolerance = 10 * 60 // 10 mins
+) => {
+  const { result } = await provider.interactionDetails(ctx.req, ctx.res);
+
+  if (!result?.login?.accountId) {
+    throw new RequestError('auth.unauthorized');
+  }
+
+  if (!result.login.ts || result.login.ts < dayjs().unix() - tolerance) {
+    const { passwordEncrypted, primaryPhone, primaryEmail } = await findUserById(
+      result.login.accountId
+    );
+
+    // No authenticated method configured for this user. Pass!
+    if (!passwordEncrypted && !primaryPhone && !primaryEmail) {
+      return;
+    }
+    throw new RequestError('auth.require_re_authentication');
```

Comment:
should guard the session regardless of the existence of these fields?

Reply:
This is the edge case that the user has no verification method specified, therefore we cannot verify that user by any means...
This can happen when the admin configured sign-in identifier as "None" in SIE v2, and have social sign-in enabled. So the user can register the account through social sign-in, and don't have to provider any identifiers (password, email or phone). And social cannot be trusted as a verification method based on our previous discussions.